src/core/loader.cpp

```diff
@@ -0,0 +1,24 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "common/io_file.h"
+#include "common/types.h"
+#include "loader.h"
+
+FileTypes DetectFileType(const std::string& filepath) {
+    if (filepath.size() == 0) // no file loaded
+    {
+        return FileTypes::Unknown;
+    }
```

Comment:
```suggestion
    // No file loaded
    if (filepath.empty()) {
        return FileTypes::Unknown;
    }
```
Prefer putting comments above in these cases so formatting doesnt break